lib/active_shipping/carriers/ups.rb

```diff
@@ -362,7 +378,32 @@ def build_shipment_request(origin, destination, packages, options={})
             end
 
             if options[:international]
-              build_international_shipment_request_options(xml, origin, destination, packages, options)
+              build_location_node(xml, 'SoldTo', options[:sold_to] || destination, options)
+
+              if origin.country_code(:alpha2) == 'US' && ['CA', 'PR'].include?(destination.country_code(:alpha2))
+                # Required for shipments from the US to Puerto Rico or Canada
+                xml.InvoiceLineTotal do
+                  total_value = packages.inject(0) {|sum, package| sum + (package.value || 0)}
+                  xml.MonetaryValue(total_value)
+                end
+              end
+
+              contents_description = packages.map {|p| p.options[:description]}.compact.join(',')
+              unless contents_description.empty?
+                xml.Description(contents_description)
+              end
+            end
+
+            xml.ShipmentServiceOptions do
+              if delivery_confirmation = options[:delivery_confirmation]
+                xml.DeliveryConfirmation do
+                  xml.DCISType(SHIPMENT_DELIVERY_CONFIRMATION_CODES[delivery_confirmation])
+                end
+              end
+
+              if options[:international]
```

Comment:
I personally would prefer this node to be called `:customs`. It's a little more inline with that I've seen on other carriers.